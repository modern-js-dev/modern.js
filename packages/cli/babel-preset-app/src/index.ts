import { createBabelChain } from '@modern-js/babel-chain';
import { applyOptionsChain } from '@modern-js/utils';
import { generate } from './generate';
import type { Options } from './type';

export type { Options };

const defaultOptions = {
  appDirectory: process.cwd(),
  metaName: 'modern-js',
  target: 'client',
  modules: false,
  useBuiltIns: 'entry',
  useModern: false,
  useLegacyDecorators: true,
  useTsLoader: false,
  lodash: {},
  styledComponents: {},
};

/* eslint-disable  no-param-reassign */
export default function (api: any, options?: Options) {
  api.cache(true);

  options = { ...(defaultOptions as Options), ...(options ?? {}) };

  const babelChain = generate(options, options.chain || createBabelChain());

  if (options.userBabelConfig) {
    return applyOptionsChain(babelChain.toJSON(), options.userBabelConfig);
  }

  return babelChain.toJSON();
}
/* eslint-enable  no-param-reassign */
