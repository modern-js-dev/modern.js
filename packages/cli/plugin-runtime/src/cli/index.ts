import {
  path,
  PLUGIN_SCHEMAS,
  createRuntimeExportsUtils,
  upath,
} from '@modern-js/utils';
import { createPlugin, usePlugins, useAppContext } from '@modern-js/core';

const useInternalDirectory = () => {
  try {
    return useAppContext().internalDirectory;
  } catch {
    return path.join(process.cwd(), 'node_modules/.modern-js');
  }
};

// eslint-disable-next-line react-hooks/rules-of-hooks
usePlugins([
  upath.normalizeSafe(require.resolve('@modern-js/plugin-state/cli')),
  upath.normalizeSafe(require.resolve('@modern-js/plugin-router/cli')),
  upath.normalizeSafe(require.resolve('@modern-js/plugin-ssr/cli')),
]);

export default createPlugin(
  () => {
    let runtimeExportsUtils: ReturnType<typeof createRuntimeExportsUtils> =
      {} as any;

    return {
      config() {
        // eslint-disable-next-line react-hooks/rules-of-hooks
        const dir = useInternalDirectory();

        runtimeExportsUtils = createRuntimeExportsUtils(dir, 'index');

        return {
          runtime: {},
          runtimeByEntries: {},
          source: {
            alias: {
              '@modern-js/runtime$': runtimeExportsUtils.getPath(),
            },
          },
        };
      },
      validateSchema() {
        return PLUGIN_SCHEMAS['@modern-js/runtime'];
      },
      addRuntimeExports() {
        const runtimePackage = path.resolve(__dirname, '../../../../');

        runtimeExportsUtils.addExport(`export * from '${runtimePackage}'`);
      },
    };
  },
  {
    name: '@modern-js/runtime',
    post: [
      '@modern-js/plugin-router',
      '@modern-js/plugin-ssr',
      '@modern-js/plugin-state',
    ],
  },
) as any;
