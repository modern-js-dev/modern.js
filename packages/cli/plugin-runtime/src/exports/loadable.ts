import loadable from '@modern-js/plugin-ssr/loadable';

export default loadable;

export * from '@modern-js/plugin-ssr/loadable';
