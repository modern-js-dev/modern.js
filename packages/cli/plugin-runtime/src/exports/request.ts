export { configure } from '@modern-js/create-request';
