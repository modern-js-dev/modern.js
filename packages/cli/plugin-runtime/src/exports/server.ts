export * from '@modern-js/bff-runtime';
export const hook = (attacher: any) => attacher;
