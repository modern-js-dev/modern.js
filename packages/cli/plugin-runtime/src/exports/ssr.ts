export * from '@modern-js/plugin-ssr';
export { useHeaders } from '@modern-js/plugin-ssr/node';
