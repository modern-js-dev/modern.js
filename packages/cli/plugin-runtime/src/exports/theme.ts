export * from '@modern-js/plugin-theme-token';
