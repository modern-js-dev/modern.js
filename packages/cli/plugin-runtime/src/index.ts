export type { RuntimeContext, TRuntimeContext } from '@modern-js/runtime-core';
export {
  createApp,
  createPlugin,
  useLoader,
  bootstrap,
  RuntimeReactContext,
  registerPrefetch,
  defineConfig,
  registerInit,
  useRuntimeContext,
} from '@modern-js/runtime-core';
