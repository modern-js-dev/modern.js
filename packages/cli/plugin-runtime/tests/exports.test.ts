/**
 * basic test of imported packages
 */
import * as PluginRuntime from '../src';
import head from '../src/exports/head';
import loadable from '../src/exports/loadable';
import * as model from '../src/exports/model';
import * as request from '../src/exports/request';
import * as router from '../src/exports/router';
import * as bff from '../src/exports/server';
import * as ssr from '../src/exports/ssr';
import * as styled from '../src/exports/styled';

describe('imported packages in plugin runtime', () => {
  it('plugin-runtime', () => {
    expect(PluginRuntime).toMatchSnapshot();
  });

  it('Helmet', () => {
    expect(head).toMatchSnapshot();
  });

  it('@loadable/component', () => {
    expect(loadable).toMatchSnapshot();
  });

  it('model', () => {
    expect(model).toMatchSnapshot();
  });

  it('request', () => {
    expect(request).toMatchSnapshot();
  });

  it('router', () => {
    expect(router).toMatchSnapshot();
  });

  it('server', () => {
    expect(bff).toBeDefined();
  });

  it('ssr', () => {
    expect(ssr).toMatchSnapshot();
  });

  it('styled', () => {
    expect(styled).toMatchSnapshot({
      version: expect.any(String),
    });
  });
});
