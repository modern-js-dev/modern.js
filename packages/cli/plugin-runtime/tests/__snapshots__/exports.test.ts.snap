// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imported packages in plugin runtime @loadable/component 1`] = `[Function]`;

exports[`imported packages in plugin runtime Helmet 1`] = `[Function]`;

exports[`imported packages in plugin runtime model 1`] = `
Object {
  "Provider": [Function],
  "autoActions": [Function],
  "createApp": [Function],
  "createStore": [Function],
  "devtools": [Function],
  "effects": [Function],
  "immer": [Function],
  "model": [Function],
  "useLocalModel": [Function],
  "useModel": [Function],
  "useStaticModel": [Function],
}
`;

exports[`imported packages in plugin runtime plugin-runtime 1`] = `
Object {
  "RuntimeReactContext": Object {
    "$$typeof": Symbol(react.context),
    "Consumer": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": [Circular],
    },
    "Provider": Object {
      "$$typeof": Symbol(react.provider),
      "_context": [Circular],
    },
    "_calculateChangedBits": null,
    "_currentRenderer": null,
    "_currentRenderer2": null,
    "_currentValue": Object {},
    "_currentValue2": Object {},
    "_threadCount": 0,
  },
  "bootstrap": [Function],
  "createApp": [Function],
  "createPlugin": [Function],
  "defineConfig": [Function],
  "registerInit": [Function],
  "registerPrefetch": [Function],
  "useLoader": [Function],
  "useRuntimeContext": [Function],
}
`;

exports[`imported packages in plugin runtime request 1`] = `
Object {
  "configure": [Function],
}
`;

exports[`imported packages in plugin runtime router 1`] = `
Object {
  "BrowserRouter": [Function],
  "HashRouter": [Function],
  "Link": Object {
    "$$typeof": Symbol(react.forward_ref),
    "propTypes": Object {
      "innerRef": [Function],
      "onClick": [Function],
      "replace": [Function],
      "target": [Function],
      "to": [Function],
    },
    "render": [Function],
  },
  "MemoryRouter": [Function],
  "NavLink": Object {
    "$$typeof": Symbol(react.forward_ref),
    "propTypes": Object {
      "activeClassName": [Function],
      "activeStyle": [Function],
      "aria-current": [Function],
      "className": [Function],
      "exact": [Function],
      "innerRef": [Function],
      "isActive": [Function],
      "location": [Function],
      "onClick": [Function],
      "replace": [Function],
      "sensitive": [Function],
      "strict": [Function],
      "style": [Function],
      "target": [Function],
      "to": [Function],
    },
    "render": [Function],
  },
  "Prompt": [Function],
  "Redirect": [Function],
  "Route": [Function],
  "Router": [Function],
  "StaticRouter": [Function],
  "Switch": [Function],
  "createBrowserHistory": [Function],
  "createHashHistory": [Function],
  "createLocation": [Function],
  "createMemoryHistory": [Function],
  "createPath": [Function],
  "generatePath": [Function],
  "locationsAreEqual": [Function],
  "matchPath": [Function],
  "parsePath": [Function],
  "useHistory": [Function],
  "useLocation": [Function],
  "useParams": [Function],
  "useRouteMatch": [Function],
  "withRouter": [Function],
}
`;

exports[`imported packages in plugin runtime server 1`] = `
Object {
  "Any": [Function],
  "Boolean": [Function],
  "Date": [Function],
  "Err": [Function],
  "Float": [Function],
  "ID": [Function],
  "Int": [Function],
  "Intersect": [Function],
  "IntersectType": [Function],
  "Json": [Function],
  "List": [Function],
  "ListType": [Function],
  "Literal": [Function],
  "LiteralType": [Function],
  "NonStrict": [Function],
  "NonStrictType": [Function],
  "Null": [Function],
  "Nullable": [Function],
  "NullableType": [Function],
  "Number": [Function],
  "ObjectType": [Function],
  "Ok": [Function],
  "PartialType": [Function],
  "ReadOnly": [Function],
  "ReadOnlyDeep": [Function],
  "ReadOnlyDeepType": [Function],
  "ReadOnlyType": [Function],
  "Record": [Function],
  "RecordType": [Function],
  "Schema": [Function],
  "Strict": [Function],
  "StrictType": [Function],
  "String": [Function],
  "Struct": [Function],
  "StructType": [Function],
  "Tuple": [Function],
  "TupleType": [Function],
  "Type": "__type",
  "Undefined": [Function],
  "Union": [Function],
  "UnionType": [Function],
  "Unknown": [Function],
  "field": [Function],
  "getInstance": [Function],
  "getSchemaCtor": [Function],
  "getSchemaCtorFields": [Function],
  "hook": [Function],
  "isFieldDescriptor": [Function],
  "isFieldDescriptors": [Function],
  "isHandler": [Function],
  "isNullableType": [Function],
  "isSchemaCtor": [Function],
  "isSchemaHandler": [Function],
  "keyof": [Function],
  "keyofObject": [Function],
  "keyofStruct": [Function],
  "match": [Function],
  "omit": [Function],
  "omitObject": [Function],
  "omitStruct": [Function],
  "partial": [Function],
  "partialObject": [Function],
  "partialStruct": [Function],
  "pick": [Function],
  "pickObject": [Function],
  "pickStruct": [Function],
  "toSchemaCtor": [Function],
  "toSchemaCtors": [Function],
}
`;

exports[`imported packages in plugin runtime ssr 1`] = `
Object {
  "NoSSR": [Function],
  "PreRender": [Function],
}
`;

exports[`imported packages in plugin runtime styled 1`] = `
Object {
  "ServerStyleSheet": [Function],
  "StyleSheetConsumer": Object {
    "$$typeof": Symbol(react.context),
    "_calculateChangedBits": null,
    "_context": Object {
      "$$typeof": Symbol(react.context),
      "Consumer": [Circular],
      "Provider": Object {
        "$$typeof": Symbol(react.provider),
        "_context": [Circular],
      },
      "_calculateChangedBits": null,
      "_currentRenderer": null,
      "_currentRenderer2": null,
      "_currentValue": undefined,
      "_currentValue2": undefined,
      "_threadCount": 0,
    },
  },
  "StyleSheetContext": Object {
    "$$typeof": Symbol(react.context),
    "Consumer": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": [Circular],
    },
    "Provider": Object {
      "$$typeof": Symbol(react.provider),
      "_context": [Circular],
    },
    "_calculateChangedBits": null,
    "_currentRenderer": null,
    "_currentRenderer2": null,
    "_currentValue": undefined,
    "_currentValue2": undefined,
    "_threadCount": 0,
  },
  "StyleSheetManager": [Function],
  "ThemeConsumer": Object {
    "$$typeof": Symbol(react.context),
    "_calculateChangedBits": null,
    "_context": Object {
      "$$typeof": Symbol(react.context),
      "Consumer": [Circular],
      "Provider": Object {
        "$$typeof": Symbol(react.provider),
        "_context": [Circular],
      },
      "_calculateChangedBits": null,
      "_currentRenderer": null,
      "_currentRenderer2": null,
      "_currentValue": undefined,
      "_currentValue2": undefined,
      "_threadCount": 0,
    },
  },
  "ThemeContext": Object {
    "$$typeof": Symbol(react.context),
    "Consumer": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": [Circular],
    },
    "Provider": Object {
      "$$typeof": Symbol(react.provider),
      "_context": [Circular],
    },
    "_calculateChangedBits": null,
    "_currentRenderer": null,
    "_currentRenderer2": null,
    "_currentValue": undefined,
    "_currentValue2": undefined,
    "_threadCount": 0,
  },
  "ThemeProvider": [Function],
  "__PRIVATE__": Object {
    "StyleSheet": [Function],
    "masterSheet": e {
      "gs": Object {},
      "names": Map {},
      "options": Object {
        "isServer": false,
        "useCSSOMInjection": false,
      },
      "server": false,
    },
  },
  "createGlobalStyle": [Function],
  "css": [Function],
  "default": [Function],
  "isStyledComponent": [Function],
  "keyframes": [Function],
  "useTheme": [Function],
  "version": "5.3.3",
  "withTheme": [Function],
}
`;
